faunadb-scala/src/load/scala/faunadb/FaunaClientFixture.scala

```diff
@@ -52,7 +54,9 @@ trait FaunaClientFixture extends SuiteMixin with BeforeAndAfterAll { self: Fixtu
     }
 
     def createFaunaClient(secret: String): Future[FaunaClient] = Future.successful {
-      rootClient.sessionClient(secret)
+      // each test gets a new client to avoid interference such as triggering max_concurrent_streams error
+      val newClient = FaunaClient(endpoint = _config("root_url"), secret = _config("root_token"))
+      newClient.sessionClient(secret)
```

Comment:
if we are creating a new client every time, no need use root_token and then call `.sessionClient` here, just pass the secret on the constructor

Reply:
Yes, this should be just:

```scala
FaunaClient(endpoint = _config("root_url"), secret = secret)
```

:)